javascript/ql/src/semmle/javascript/TypeScript.qll

```diff
@@ -2610,6 +2614,30 @@ class CallSignatureType extends @signature_type {
    * Gets the name of a parameter of this signature.
    */
   string getAParameterName() { result = getParameterName(_) }
+
+  /**
+   * Holds if this signature declares a rest parameter, such as `(x: number, ...y: string[])`.
+   */
+  predicate hasRestParameter() { signature_rest_parameter(this, _) } // TODO
```

Comment:
Seems like you forgot to remove the `// TODO`, because the line looks fine.

Reply:
Replace with `exists(getRestParameterArrayType())` for readability?

Reply:
I prefer to use the database relation so these classes don't have unexpected dependencies. For instance, this predicate would come to depend on the charpred of `PlainArrayType`, which itself has some non-trivial dependencies. Nothing bad would happen in this case, but that's my reason for doing it.